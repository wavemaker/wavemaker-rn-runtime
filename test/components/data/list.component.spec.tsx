import React, { ReactNode } from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import WmList from '@wavemaker/app-rn-runtime/components/data/list/list.component';
import WmListProps from '@wavemaker/app-rn-runtime/components/data/list/list.props';
import { render } from '@testing-library/react-native';
describe('Test List component', () => {
  interface ListItem {
    id: number;
    name: string;
    category?: string;
  }

  const mockData: ListItem[] = [
    { id: 1, name: 'Item 1' },
    { id: 2, name: 'Item 2' },
    { id: 3, name: 'Item 3' }
  ];

  //test id
  const testID = "testList_flat_list"

  const createProps = (overrides?: Partial<WmListProps>): WmListProps => {
    const baseProps = new WmListProps();
    return {
      ...baseProps,
      name: 'testList',
      direction: 'horizontal',
      dataset: mockData,
      renderItem: (item: ListItem): JSX.Element => <div>{item.name}</div>,
      ...overrides
    };
  };
  test('Check validity of sample component', () => {
    const props = createProps();
    const tree = render(<WmList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should show horizontal scrollbar by default', () => {
    const props = createProps();
    const { getByTestId } = render(<WmList {...props} />);
    const flatList = getByTestId(testID);
    expect(flatList.props.showsHorizontalScrollIndicator).toBe(true);
  });

  test('should hide horizontal scrollbar when hidehorizontalscrollbar is true', () => {
    const props = createProps({ hidehorizontalscrollbar: true });
    const { getByTestId } = render(<WmList {...props} />);
    const flatList = getByTestId(testID);
    expect(flatList.props.showsHorizontalScrollIndicator).toBe(false);
  });

  test('should show horizontal scrollbar when hidehorizontalscrollbar is false', () => {
    const props = createProps({ hidehorizontalscrollbar: false });
    const { getByTestId } = render(<WmList {...props} />);
    const flatList = getByTestId(testID);
    expect(flatList.props.showsHorizontalScrollIndicator).toBe(true);
  });

  test('should not affect vertical list scrollbar', () => {
    const props = createProps({
      direction: 'vertical',
      hidehorizontalscrollbar: true
    });

    const { getByTestId } = render(<WmList {...props} />);
    const flatList = getByTestId(testID);

    expect(flatList.props.showsHorizontalScrollIndicator).toBeUndefined();
  });
});