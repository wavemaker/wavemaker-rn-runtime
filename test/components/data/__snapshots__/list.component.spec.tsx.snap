// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List component Check validity of sample component 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={{}}
  >
    <View
      style={
        {
          "alignSelf": "center",
          "color": "#666666",
          "flex": 1,
          "fontSize": 16,
          "lineHeight": 18,
        }
      }
      testID="non_animatableView"
    >
      <View>
        <Text
          accessibilityLabel="No data found"
          accessibilityRole="text"
          accessible={true}
          ellipsizeMode="tail"
          style={
            {
              "color": "#888888",
              "fontFamily": "Roboto",
              "fontSize": 16,
              "userSelect": "text",
            }
          }
          testID="test_List_emptymsg_caption"
        >
          No data found
        </Text>
      </View>
    </View>
  </View>
</View>
`;
